src/transparent.rs

```diff
@@ -279,147 +318,328 @@ pub fn zstd_enable_transparent<'a>(ctx: &Context) -> anyhow::Result<ToSqlOutput<
+            trg_name = escape_sqlite_identifier(&trigger_name),
+            view_name = escape_sqlite_identifier(&table_name),
+            backing_table = escape_sqlite_identifier(&internal_table_name),
+            upd_col = escape_sqlite_identifier(&col.name),
+            update = update,
             primary_key_condition = primary_key_condition
         );
-        log::debug!("[run] {}", &deletetrigger_query);
-        db.execute(&deletetrigger_query, params![])
-            .context("could not create delete trigger")?;
+        log::debug!("[run] {}", &updatetrigger_query);
+        db.execute(&updatetrigger_query, params![])
+            .with_context(|| format!("Could not create update of {} trigger", col.name))?;
     }
-    {
-        for col in &columns_info {
-            let update = if col.name == to_compress_column.name {
-                format!(
-                    "{col} = zstd_compress_col(new.{col}, {lvl}, {dictcol}, {compact})",
-                    col = escape_sqlite_identifier(&col.name),
-                    lvl = config.compression_level,
-                    dictcol = escape_sqlite_identifier(&dict_id_column_name),
-                    compact = COMPACT
-                )
-            } else {
-                format_sqlite!("{} = new.{}", &col.name, &col.name)
-            };
-            // update triggers
-            let updatetrigger_query = format!(
-                "
-                create trigger {trg_name}
-                    instead of update of {upd_col} on {view_name}
-                    for each row
-                    begin
-                        update {backing_table} set {update} where {primary_key_condition};
-                    end;
-                ",
-                trg_name = escape_sqlite_identifier(&format!(
-                    "{}_update_{}_trigger",
-                    table_name, col.name
-                )),
-                view_name = escape_sqlite_identifier(&table_name),
-                backing_table = escape_sqlite_identifier(&new_table_name),
-                upd_col = escape_sqlite_identifier(&col.name),
-                update = update,
-                primary_key_condition = primary_key_condition
-            );
-            log::debug!("[run] {}", &updatetrigger_query);
-            db.execute(&updatetrigger_query, params![])
-                .with_context(|| format!("Could not create update of {} trigger", col.name))?;
-        }
+    Ok(())
+}
+
+fn get_configs(
+        db: &rusqlite::Connection
+    ) -> Result<Vec<TransparentCompressConfig>, anyhow::Error> {
+    let table_count: u32 = db
+        .query_row(
+            "select count(`type`) 
+            from sqlite_master 
+            where name = '_zstd_configs' and type = 'table'",
+            params![],
+            |r| r.get(0),
+        )
+        .with_context(|| format!("Could not get table information for _zstd_configs"))?;
+    if table_count == 0 {
```

Comment:
i feel like the `check_table_exists` should be changed to only have the table_name argument, return a boolean and used here as well?

Reply:
Sure) it is much better